src/modules/75-cd/components/PipelineSteps/Common/Terraform/TerraformInterfaces.ts

```diff
@@ -306,17 +306,22 @@ export const onSubmitTerraformData = (values: any): TerraformData => {
   const envVars = get(values.spec, `${fieldPath}.spec.environmentVariables`)
   const targets = get(values.spec, `${fieldPath}.spec.targets`) as MultiTypeInputType
   const cmdFlags = get(values.spec, `${fieldPath}.commandFlags`)
-  const processCmdFlags = cmdFlags?.map((commandFlag: TerraformCliOptionFlag) =>
-    commandFlag.commandType && commandFlag.flag
-      ? {
-          commandType: commandFlag.commandType,
-          flag: commandFlag.flag
-        }
-      : {
-          commandType: '',
-          flag: ''
-        }
-  )
+
+  const processCmdFlags = (): TerraformCliOptionFlag[] | undefined => {
+    if (cmdFlags?.length && cmdFlags[0].commandType) {
+      return cmdFlags?.map((commandFlag: TerraformCliOptionFlag) =>
+        commandFlag.commandType && commandFlag.flag
+          ? {
+              commandType: commandFlag.commandType,
+              flag: commandFlag.flag
+            }
+          : {
+              commandType: '',
+              flag: ''
+            }
+      )
+    }
+  }
```

Comment:
```suggestion
  const processCmdFlags = (): TerraformCliOptionFlag[] | undefined => {
  if (cmdFlags?.length && cmdFlags[0].commandType) {
    return cmdFlags.map((commandFlag: TerraformCliOptionFlag) => ({
      commandType: commandFlag.commandType,
      flag: defaultTo(commandFlag?.flag , ""),
    }));
  }
};

```
map function will be called if commandFlag.commandType exists, so we can omit ternary